packages/app/src/components/Admin/Notification/ManageGlobalNotification.jsx

```diff
@@ -1,330 +1,336 @@
-      notifyToType: globalNotification.__t || 'mail',
-      emailToSend: globalNotification.toEmail || '',
-      slackChannelToSend: globalNotification.slackChannels || '',
-      triggerEvents: new Set(globalNotification.triggerEvents),
-    };
-
-    this.submitHandler = this.submitHandler.bind(this);
-  }
-
-  onChangeTriggerPath(inputValue) {
-    this.setState({ triggerPath: inputValue });
-  }
-
-  onChangeNotifyToType(notifyToType) {
-    this.setState({ notifyToType });
-  }
-
-  onChangeEmailToSend(inputValue) {
-    this.setState({ emailToSend: inputValue });
-  }
-
-  onChangeSlackChannelToSend(inputValue) {
-    this.setState({ slackChannelToSend: inputValue });
-  }
-
-  onChangeTriggerEvents(triggerEvent) {
-    const { triggerEvents } = this.state;
+const ManageGlobalNotification = (props) => {
+
+  // constructor() {
+  //   super();
+
+  //   const globalNotification = null;
+  //   // try {
+  //   //   globalNotification = JSON.parse(document.getElementById('admin-global-notification-setting').getAttribute('data-global-notification'));
+  //   // }
+  //   // catch (err) {
+  //   //   // toastError(err);
+  //   //   logger.error(err);
+  //   // }
+
+  //   this.state = {
+  //     // globalNotificationId: globalNotification._id || null,
+  //     globalNotificationId: null,
+  //     // triggerPath: globalNotification.triggerPath || '',
+  //     triggerPath: '',
+  //     // notifyToType: globalNotification.__t || 'mail',
+  //     notifyToType: 'mail',
+  //     // emailToSend: globalNotification.toEmail || '',
+  //     emailToSend: '',
+  //     // slackChannelToSend: globalNotification.slackChannels || '',
+  //     slackChannelToSend: '',
+  //     // triggerEvents: new Set(globalNotification.triggerEvents),
+  //     triggerEvents: new Set(),
+  //   };
+
+  //   this.submitHandler = this.submitHandler.bind(this);
+  // }
+
+  const [globalNotificationId, setGlobalNotificationId] = useState(null);
+  const [triggerPath, setTriggerPath] = useState('');
+  const [notifyToType, setNotifyToType] = useState('mail');
+  const [emailToSend, setEmailToSend] = useState('');
+  const [slacChannelToSend, setSlackChannelToSend] = useState('');
+  const [triggerEvents, setTriggerEvents] = useState(new Set());
+  const [retrieveError] = useState(null);
+
+
+  // onChangeTriggerPath(inputValue) {
+  //   this.setState({ triggerPath: inputValue });
+  // }
+
+  // onChangeNotifyToType(notifyToType) {
+  //   this.setState({ notifyToType });
+  // }
+
+  // onChangeEmailToSend(inputValue) {
+  //   this.setState({ emailToSend: inputValue });
+  // }
+
+  // onChangeSlackChannelToSend(inputValue) {
+  //   this.setState({ slackChannelToSend: inputValue });
+  // }
+
+  const onChangeTriggerEvents = (triggerEvent) => {
 
     if (triggerEvents.has(triggerEvent)) {
       triggerEvents.delete(triggerEvent);
-      this.setState({ triggerEvents });
+      setTriggerEvents({ triggerEvents });
```

Comment:
これは と class の setState と functional componentの の setXXXの 違いですね

`setState`は複数の変数の値を書き換えられるのでオブジェクトになります。

他のsetHoge も修正お願いします。

```suggestion
      setTriggerEvents({ triggerEvents });
```

Reply:
上記のように修正済み。